src/components/AddressSearch.js

```diff
@@ -109,12 +107,15 @@ const AddressSearch = (props) => {
                 label: props.label,
                 containerStyles: props.containerStyles,
                 errorText: props.errorText,
+                value: props.value,
                 onChangeText: (text) => {
-                    const isTextValid = !_.isEmpty(text) && _.isEqual(text, props.value);
-
-                    // Ensure whether an address is selected already or has address value initialized.
-                    if (!_.isEmpty(googlePlacesRef.current.getAddressText()) && !isTextValid) {
-                        saveLocationDetails({});
+                    // We use `skippedFirstOnChangeText` to work around a feature of the library:
+                    // The library is calling onChangeText with '' at the start and we don't need this
+                    // https://github.com/FaridSafi/react-native-google-places-autocomplete/blob/47d7223dd48f85da97e80a0729a985bbbcee353f/GooglePlacesAutocomplete.js#L148
```

Comment:
Maybe we should move this comment to where we declare `skippedFirstOnChangeText`.